### ISSUE TITLE
Mapping to object of the same type doesn't work

### ISSUE THREAD

**Issue by bzuillsmith** [expressmapper/ExpressMapper#32]
I have some complex entities that I need to copy but exclude several Object-type properties. However, it wasn't working for me. Then I tried this simple example and found it doesn't work. It appears I get back the ```default(DestinationType)```
```C#
var mapper = new MappingService();
mapper.Register<PartType, PartType>();

mapper.Compile();

// The only property is "Type"
var partType = new PartType()
{
  Type = "Test" 
};

var partTypeCopy = mapper.Map<PartType, PartType>(partType);
Assert.AreEqual(partTypeCopy.Type, partType.Type); // Fails here
```

**Comment by anisimovyuriy** [comment]
Dear bzuillsmith,

Thanks a lot for th input. You are right I've just checked it and yes it ignores the same for some reason. So this is the bug and we'll fix it ASAP.

**Comment by Tornhoof** [comment]
Any ETA in that 1.1 Version?
I could use that functionality too.

**Comment by anisimovyuriy** [comment]
Tornhoof,

September the 15th!

**Comment by anisimovyuriy** [comment]
Fixed and will be included 1.1 release
